web/ui/src/features/river-js/RiverValue.tsx

```diff
@@ -1,30 +1,56 @@
-import React, { FC, Fragment, ReactElement } from 'react';
+import React, { FC, Fragment } from 'react';
 
 import { ObjectField, Value, ValueType } from './types';
 
 import styles from './RiverValue.module.css';
 
 export interface RiverValueProps {
   value: Value;
+  name?: string;
+  nthChild?: number;
 }
 
+// TO track the length of the object key value pair
+const DIVISION_LENGTH = 20;
+
 /**
  * RiverValue emits a paragraph which represents a River value.
  */
-export const RiverValue: FC<RiverValueProps> = (props) => {
+export const RiverValue: FC<RiverValueProps> = ({ value, name, nthChild }) => {
   return (
-    <p className={styles.value}>
-      <ValueRenderer value={props.value} indentLevel={0} />
-    </p>
+    <div className={styles.value}>
+      <ValueRenderer value={value} nthChild={nthChild} name={name} />
+    </div>
   );
 };
 
-type valueRendererProps = RiverValueProps & {
-  indentLevel: number;
-};
+const ValueRenderer: FC<RiverValueProps> = ({ value, name, nthChild }) => {
+  const backgroundColor = nthChild && nthChild % 2 === 1 ? '#f4f5f5' : 'white';
+
+  /**
+   * Renderer for the river format for target and label object
+   */
+  const renderGrid = (partition: ObjectField[]) => {
+    // const gridTemplateColumns = '10% 1% 69%';
+
+    let maxLength = 0;
 
-const ValueRenderer: FC<valueRendererProps> = (props) => {
-  const value = props.value;
+    partition.forEach(({ key }) => {
+      return key.length > maxLength ? (maxLength = key.length) : maxLength;
+    });
+
+    const gridTemplateColumns = maxLength < DIVISION_LENGTH ? '10% 1% 89%' : undefined;
+
+    return partition.map(({ key, value }) => {
+      return (
+        <div key={key} className={styles['grid-layout']} style={{ backgroundColor: backgroundColor, gridTemplateColumns }}>
```

Comment:
Is there a reason that each of the attributes is put in the DOM as a separate grid?

If you had each attribute as a separate row in the grid, then you could set `grid-template-columns` to `max-content auto auto` and it would fluidly scale as expected, but because each attribute is a different grid, we're forced to do the `10% 1% 89%` check, where even the keys could overflow if they're long enough: 

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/630212/214149797-d001f8ba-e335-4612-9d5c-c67ffec0f14f.png">

(Notice the overflowing on `__meta_kubernetes_pod_annotationpresent_scheduler_alpha_kubernetes_io_critical_pod`)

Reply:
If we set a `max-content auto auto` then the width will depend on the content length, which will cause 2 things to happen: 
1. the line up will be messed up as the content length will be different
2. the content will not wrap which will cause an overflow

Reply:
> Is there a reason that each of the attributes is put in the DOM as a separate grid?

Oh I see what you mean now. I didn't realize I was setting the grid layout inside the loop. My bad. Updating it now. Nice catch thanks!